crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -222,7 +222,12 @@ impl GeneratedAcir {
         witness: Witness,
         num_bits: u32,
     ) -> Result<(), AcirGenError> {
-        if num_bits == FieldElement::max_num_bits() {
+        // We class this as an error because users should instead
+        // do `as Field`.
+        // TODO: Since these are compiler time knowns,
+        // TODO this could be changed into a compiler panic and enforced
+        // TODO on the frontend instead.
```

Comment:
I'd be ok with this. Want a separate issue opened?

Reply:
Yes please

Reply:
I've created an issue for this